sdk/mobile-center-push/src/main/java/com/microsoft/azure/mobile/push/Push.java

```diff
@@ -0,0 +1,256 @@
+
+    /**
+     * Init.
+     */
+    private Push() {
+        mFactories = new HashMap<>();
+        mFactories.put(PushInstallationLog.TYPE, new PushInstallationLogFactory());
+    }
+
+    /**
+     * Get shared instance.
+     *
+     * @return shared instance.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public static synchronized Push getInstance() {
+        if (sInstance == null) {
+            sInstance = new Push();
+        }
+        return sInstance;
+    }
+
+    @VisibleForTesting
+    static synchronized void unsetInstance() {
+        sInstance = null;
+    }
+
+    /**
+     * Check whether Push service is enabled or not.
+     *
+     * @return <code>true</code> if enabled, <code>false</code> otherwise.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public static boolean isEnabled() {
+        return getInstance().isInstanceEnabled();
+    }
+
+    /**
+     * Enable or disable Push service.
+     *
+     * @param enabled <code>true</code> to enable, <code>false</code> to disable.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public static void setEnabled(boolean enabled) {
+        getInstance().setInstanceEnabled(enabled);
+    }
+
+    /**
+     * Change the GSM sender ID.
+     *
+     * @param senderId GSM sender ID
+     */
+    @SuppressWarnings({"WeakerAccess", "SameParameterValue"})
+    public static void setSenderId(String senderId) {
+        getInstance().setInstanceSenderId(senderId);
+    }
+
+    /**
+     * Enqueue a push installation log.
+     *
+     * @param pushToken the push token value
+     */
+    private void enqueuePushInstallationLog(@NonNull String pushToken) {
+        if (isInactive())
+            return;
+        if (mPushTokenSent)
+            return;
+
+        PushInstallationLog log = new PushInstallationLog();
+        log.setPushToken(pushToken);
+        mChannel.enqueue(log, PUSH_GROUP);
+        mPushTokenSent = true;
+    }
+
+    /**
+     * Handle push token update success.
+     *
+     * @param pushToken the push token value
+     */
+    @VisibleForTesting
+    synchronized void handlePushToken(@NonNull String pushToken) {
+        mPushToken = pushToken;
+        MobileCenterLog.debug(LOG_TAG, "Push token: " + mPushToken);
+        PreferencesStorage.putString(PREFERENCE_KEY_PUSH_TOKEN, mPushToken);
+        enqueuePushInstallationLog(mPushToken);
+    }
+
+    private synchronized void updatePushToken() {
+        if (isInactive())
+            return;
+        if (mPushTokenSent)
+            return;
+
+        if (mPushToken != null) {
+            enqueuePushInstallationLog(mPushToken);
+        } else if (mSenderId != null) {
+            final PushTokenTask pushTokenTask = new PushTokenTask();
+            try {
+                pushTokenTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+            } catch (final RejectedExecutionException e) {
+                MobileCenterLog.error(LOG_TAG, "THREAD_POOL_EXECUTOR saturated", e);
```

Comment:
Please add a TODO comment here, so that we use ASyncTaskUtils when its gets merged in the future.